src/adc.rs

```diff
@@ -456,6 +458,48 @@ impl_adc!(ADC1: adc_unit_t_ADC_UNIT_1);
 #[cfg(not(any(esp32c2, esp32h2, esp32c5, esp32c6, esp32p4)))] // TODO: Check for esp32c5 and esp32p4
 impl_adc!(ADC2: adc_unit_t_ADC_UNIT_2);
 
+/// One-shot ADC module
+/// Example: reading a value form a pin and printing it on the terminal
+/// ```
+/// use std::thread;
+/// use std::time::Duration;
+
+/// #[cfg(not(any(feature = "adc-oneshot-legacy", esp_idf_version_major = "4")))]
```

Comment:
should probabbly also remove this cfg(not(any .. as a user would only look at this if the module itself is visible in the first place,

Reply:
Opps, sorry. Should have looked closer. All `cfg` lines should be removed now. And I did a quick check to make the examples look somewhat similar too.